### PR TITLE
feat(component): add sub-rows indicators for worksheet

### PR DESCRIPTION
**Pull request by bc-apostoliuk**
## What?
1. Add sub-rows indicators for the worksheet.
2. Changes in the worksheet according to the new guidelines.

## Why?
The guidelines is updated.

## Screenshots/Screen Recordings
Screenshot 2023-01-25 at 14 01 20

## Testing/Proof

Locally.

### CODE REVIEW

**Review by vdm-kr — APPROVED** [review]
Looks good overall. The only thing, can we decrease the first column's left padding to 8px?

**Comment by bc-apostoliuk** [comment]
> Looks good overall. The only thing, can we decrease the first column's left padding to 8px?

@vdm-kr  Yes, we can. I have pushed these changes.
Screenshot 2023-01-25 at 14 59 32

**Review by jorgemoya — APPROVED** [review]
Looks cool!

File: packages/big-design/src/components/Worksheet/Worksheet.tsx [hunk 1]
```diff
@@ -172,6 +172,7 @@ const InternalWorksheet = typedMemo(
       <UpdateItemsProvider items={rows}>
         <StyledBox>
           <Table
+            hasExpandableRows={!!expandableRows}
```
Comment (MariaJose) [inline packages/big-design/src/components/Worksheet/Worksheet.tsx:175]: `Boolean(expandableRows)` 🍹

File: packages/big-design/src/components/Worksheet/editors/ToggleEditor/ToggleEditor.tsx [hunk 2]
```diff
@@ -25,29 +24,29 @@ const InternalToggleEditor = ({ rowId, toggle }: ToggleEditorProps) => {
...
-          variant="subtle"
-        />
-      )}
-    </Flex>
-  ) : null;
+  if (!isExpandable) {
+    return null;
+  }
+
+  return hasExpanded ? (
```
Comment (MariaJose) [inline packages/big-design/src/components/Worksheet/editors/ToggleEditor/ToggleEditor.tsx:31]: probably if we use another if in here might look cleaner 🍹 
````tsx
if (hasExpanded) {
  return ( 
  <StyledExpandButton
      iconOnly={<ExpandMoreIcon color="secondary60" />}
      onClick={() => {
        onToggle(true);
      }}
      title="toggle row expanded"
      variant="subtle"
    />
  )
}

return (
<StyledExpandButton
      iconOnly={<ChevronRightIcon color="secondary60" />}
      onClick={() => {
        onToggle(false);
      }}
      title="toggle row expanded"
      variant="subtle"
    />
)
```

**Review by MariaJose — APPROVED** [review]

File: packages/big-design/src/components/Worksheet/editors/ToggleEditor/ToggleEditor.tsx (hunk 2 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Worksheet/editors/ToggleEditor/ToggleEditor.tsx:31]: @MariaJose Thanks, done))

File: packages/big-design/src/components/Worksheet/Worksheet.tsx (hunk 1 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Worksheet/Worksheet.tsx]: Done

**Review by golcinho — APPROVED** [review]

**Pull request merged**